src/server/routes/htw.js

```diff
@@ -93,6 +94,65 @@ export function setupHtw(App) {
     )
   })
 
+  App.express.get('/api/wechall/validate', async (req, res) => {
+    const username = req.query.username?.toString() ?? ''
+    const email = req.query.email?.toString() ?? ''
+    const authkey = req.query.authkey?.toString() ?? ''
+
+    if (authkey !== secrets('secret_wechall_authkey')) {
+      res.send('0')
+      return
+    }
+
+    const token = generateWeChallToken(username)
+
+    if (email === token) {
+      res.send('1')
+    } else {
+      res.send('0')
+    }
+  })
+
+  App.express.get('/api/wechall/score', async (req, res) => {
+    const username = req.query.username?.toString() ?? ''
+    const authkey = req.query.authkey?.toString() ?? ''
+
+    /*if (authkey !== secrets('secret_wechall_authkey')) {
+      res.send('error')
+      return
+    }*/
+
+    const user = await App.db.models.User.findOne({
+      where: { name: username },
+    })
+
+    if (!user) {
+      res.send('error')
+      return
+    }
+
+    const betterThanMe = await App.db.models.User.count({
+      where: {
+        [Op.or]: [{ score: { [Op.gt]: user.score } }],
+      },
+    })
+    const rank = user.score == 0 ? -1 : betterThanMe + 1
+    const sum = await App.db.models.User.count({
+      where: { score: { [Op.gt]: 0 } },
+    })
+
+    const maxScore = await App.db.models.User.max('score')
+
+    const cids = App.challenges.data.filter((c) => !c.noScore).map((c) => c.id)
+    const solved = await App.db.models.Solution.count({
+      where: { UserId: user.id, cid: cids },
+    })
+
+    res.send(
+      `${username}:${rank}:${user.score}:${sum}:${solved}:${cids.length}:${maxScore}`
```

Comment:
You swapped `maxscore` and `usercount` (aka `sum`)
Otherwise _scoreurl_ seems to look OK

Reply:
true, thank you for pointing this out